broadinstitute/consent
# NPE Fix for un-configured user access

NPE fix for when an un-configured user tries to hit any basic auth endpoint. Discovered when testing for https://broadinstitute.atlassian.net/browse/GAWB-1676 on alpha. The priority for this is relatively minor since the error case still triggers an error ... just the wrong kind of error.

No jira ticket for this.

### Testing
Testing requires that you have some kind of user role in DUOS. The authentication handler for both `/api/*` calls and `/swagger/*` are now routed through the same auth filter.
1. When running, authenticate on the swagger page (https://local.broadinstitute.org/swagger/).
2. In the swagger UI, test out any valid API call such as the consent summary endpoint.
3. Copy the curl call and test that on the command line. 
4. Both should work equally well and all swagger resources should load.
5. Additionally, one can test the compliance portal by pointing it to your local instance and try exporting a consent to DUOS. This uses the `/basic/*` endpoint and hits the other authentication filter that's in use.

---
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
